### PR TITLE
Migrate smart contracts extensions to hardhat

### PR DESCRIPTION
**Pull request by clemsos**
# Description

- [x] remove truffle 
- [x] cleanup deps
- [x] fix tests  

# Issues

Relates to discussion in #7446 about upgrading packages

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: smart-contract-extensions/test/hooks/discountCodeHook.js [hunk 1]
```diff
@@ -214,7 +214,7 @@ contract('DiscountCodeHook', accounts => {
           from: keyBuyer,
         }
       ),
-      'ONLY_LOCK_MANAGER.'
+      "VM Exception while processing transaction: reverted with reason string 'ONLY_LOCK_MANAGER'"
```
Comment (clemsos) [inline smart-contract-extensions/test/hooks/discountCodeHook.js:217]: one thing I dont get here is why this was throwing, when others `reverts` test were not, such as 

https://github.com/unlock-protocol/unlock/blob/065eb9ef5488eecb30fc58c060e21d5af1d92b54/smart-contract-extensions/test/keyPurchaser.js#L44-L52

Comment (julien51) [reply smart-contract-extensions/test/hooks/discountCodeHook.js:217]: Oh well.

File: smart-contract-extensions/test/uniswapAndCall.js [hunk 2]
```diff
@@ -16,11 +16,16 @@ async function createAndFundExchange(uniswap, token, tokenOwner) {
...
   await token.approve(exchange.address, -1, { from: tokenOwner })
+  
+  const blockNumber = await web3.eth.getBlockNumber()
+  const latestBlock = await web3.eth.getBlock(blockNumber)
+
   await exchange.addLiquidity(
     '1',
     '1000000000000000000000000',
-    Math.round(Date.now() / 1000) + 60,
+    latestBlock.timestamp + 60,
```
Comment (julien51) [inline smart-contract-extensions/test/uniswapAndCall.js:28]: 👍

**Review by julien51 — APPROVED** [review]

**Comment by julien51** [comment]
Nice!

**Comment by julien51** [comment]
i think once merged we can trigger a rebase of https://github.com/unlock-protocol/unlock/pull/6870

**Comment by clemsos** [comment]
ok will try the rebase then

**Comment by julien51** [comment]
oh you can just click the checkbox :)

**Comment by clemsos** [comment]
oh great !

**Pull request merged**